src/docfx/build/redirection/RedirectionProvider.cs

```diff
@@ -201,22 +231,35 @@ private static RedirectionItem[] LoadRedirectionModel(ErrorBuilder errors, strin
                             });
                         }
                     }
-
-                    return results.OrderBy(item => item.RedirectUrl.Source).ToArray();
                 }
             }
 
-            return Array.Empty<RedirectionItem>();
+            return results.OrderBy(item => item.RedirectUrl.Source).ToArray();
         }
 
-        private static IEnumerable<string> ProbeRedirectionFiles(string docsetPath, Repository? repository)
+        private static IEnumerable<(string redirectionFile, bool IsSubRedirectionFile)> ProbeRedirectionFiles(
+            string docsetPath,
+            Repository? repository,
+            HashSet<string> redirectionFiles)
         {
-            yield return Path.Combine(docsetPath, "redirections.yml");
-            yield return Path.Combine(docsetPath, "redirections.json");
+            yield return (Path.Combine(docsetPath, "redirections.yml"), true);
+            yield return (Path.Combine(docsetPath, "redirections.json"), true);
 
             if (repository != null)
             {
-                yield return Path.Combine(repository.Path, ".openpublishing.redirection.json");
+                yield return (Path.Combine(repository.Path, ".openpublishing.redirection.json"), false);
+
+                // redirection files registered in .openpublishing.publish.config.json
+                foreach (var item in redirectionFiles)
+                {
+                    if (item.Equals(".openpublishing.redirection.json")
+                        || item.Equals(Path.GetRelativePath(repository.Path, Path.Combine(docsetPath, "redirections.yml")))
+                        || item.Equals(Path.GetRelativePath(repository.Path, Path.Combine(docsetPath, "redirections.json"))))
+                    {
+                        continue;
+                    }
+                    yield return (Path.Combine(repository.Path, item), true);
```

Comment:
how about combining repo root here directly? Then no need to return `IsSubRedirectionFile`

Reply:
I think it's easier to understand the code for others.

Reply:
Consider having 2 methods:
- ProbeMainRedirectionFiles: same implementation as before
- ProbeSubRedirectionFiles: probe from config

Reply:
I don't think there is a need to have 2 methods.